src/pymmcore_widgets/_mda/_time_plan_widget.py

```diff
@@ -261,16 +264,21 @@ def set_state(self, t_plan: dict) -> None:
             ) `interval` key is not a `timedelta` object, it will be converted to a
             timedelta object and will be considered as expressed in seconds.
         """
-        self._clear()
-
         tp = MDASequence(time_plan=t_plan).time_plan
         if tp is None:
+            # should we raise/warn here?
             return
-        phases = tp.phases if isinstance(tp, MultiPhaseTimePlan) else [tp]
-        for phase in phases:
-            if not isinstance(phase, TIntervalLoops):
-                raise ValueError("Time dicts must have both 'interval' and 'loops'.")
-            self._create_new_row(interval=phase.interval, loops=phase.loops)
+        with signals_blocked(self):
+            self._clear()
+            phases = tp.phases if isinstance(tp, MultiPhaseTimePlan) else [tp]
+            with signals_blocked(self):
```

Comment:
is this duplicate usage of `signals_blocked` intentional @ianhi?

Reply:
i deleted and tests still pass... so will go with it.  Let me know if i'm mistaken

Reply:
this was correct to remove. I must've missed the internal one when i came back to include the `clear` inside of the blocker

Reply:
thanks for confirming!